cEP-0022.md

```diff
@@ -0,0 +1,160 @@
+# Quickstart Green Mode
+
+| Metadata |                                               |
+| -------- | --------------------------------------------- |
+| cEP      | 22                                            |
+| Version  | 1.0                                           |
+| Title    | Quickstart Green Mode                         |
+| Authors  | Ishan Srivastava <mailto:ishan.srt@gmail.com> |
+| Status   | Implementation Due                            |
+| Type     | Feature                                       |
+
+# Abstract
+
+This document describes what is the **Green Mode** for `coala-quickstart` and how
+will this be implemented.
+
+This cEP describes new type of config files which will be generated by
+`coala-quickstart`, how will they be different from the ones currently being
+created, how is this helpful and the action-plan for adding this feature.
+
+# Introduction
+
+Currently `coala-quickstart` generates config files which point out too many
+errors in the code-base of the project. It also asks too many questions
+while generating the config file. The config files are not that specific to
+the project and the errors pointed out by coala using generated config files
+are generally undesirable for the project owners.
+
+Project owners need a configuration file which *agrees* with their codestyle, is
+as specific as possible i.e. can point out maximum errors with least amount
+of manual edits in the future, for future inconsistencies in commits.
+Many organizations don't follow specific code styles like `PEP8`, instead they
+follow their own standards. We need to adapt to those standards with least
+user interaction and produce config files which generate absolutely no errors
+in their codebase. This will lead to easier adoption of coala across communities
+as it is going to result in an easy mergeable PR in the orgs.
```

Comment:
> an easy mergeable PR in the orgs

clarify this.

this 'PR' is referring to an 'adding .coafile and CI to enforce it' PR.

see `https://github.com/processing/processing/pull/4781` for example, which isnt merged despite being quite intelligent decisions.

Reply:
The project owner seems confused about what that PR is for